joomla/joomla-cms
# [RFC] [com_installer updatesites] Do not allow to enable/disable/delete a protected extension update site

### Summary of Changes

This makes com_installer update sites respect the protected status of the extensions, ie, when the extension is protected it will not be able to be enable/disable/delete their update sites.

TODO:
- [ ] update sql to enable all protected extension update sites to have them etternaly disabled
- [x] figure out out to solve the core disabling update sites when not reachable

### Testing Instructions

0. Code review
1. Apply patch
2. Check in extensions manage which extensions are protected.
3. Go to extensions -> manage -> update sites and check the update sites of those extensions are now protected (cannot be enabled or disabled nor deleted).
4. Try to disable/enable/delete update sites of protected extensions. You should NOT be able to do it.
5. Try to disable/enable/delete update sites of unprotected extensions. You should be able to do it.

### Documentation Changes Required

None.
